src/station_cmd.cpp

```diff
@@ -1376,6 +1478,33 @@ CommandCost CmdBuildRailStation(DoCommandFlags flags, TileIndex tile_org, RailTy
 	/* Make sure the area below consists of clear tiles. (OR tiles belonging to a certain rail station) */
 	StationID est = StationID::Invalid();
 	std::vector<Train *> affected_vehicles;
+
+	const StationSpec *statspec = StationClass::Get(spec_class)->GetSpec(spec_index);
+
+	TileIndexDiff tile_delta = TileOffsByAxis(axis); // offset to go to the next platform tile
+	TileIndexDiff track_delta = TileOffsByAxis(OtherAxis(axis)); // offset to go to the next track
+	std::vector<uint8_t> layout_buffer;
+	layout_buffer.resize(numtracks * plat_len);
+	//TempBufferST<uint8_t> layout_buffer(numtracks * plat_len);
+	GetStationLayout(&layout_buffer[0], numtracks, plat_len, statspec);
```

Comment:
This is all done later on in this function. Instead of duplicating, it should be moved from where it is to earlier where you need it.